lib/adf_info.py

```diff
@@ -567,6 +616,62 @@ def add_diag_var(self, var_str):
             self.__diag_var_list.append(var_str)
         #End if
 
+    #########
+
+    #Utility function to grab climo years from pre-made time series files:
+    def get_climo_yrs_from_ts(self, input_ts_loc, case_name):
+        """
+        Grab start and end climo years if none are specified in config file
+        for pre-made time series file(s)
+
+        Return
+        ------
+          - start year
+          - end year
+        """
+
+        #Grab variable list
+        var_list = self.__diag_var_list.copy()
+
+        #Create "Path" objects:
+        input_location  = Path(input_ts_loc)
+
+        #Check that time series input directory actually exists:
+        if not input_location.is_dir():
+            errmsg = f"Time series directory '{input_ts_loc}' not found.  Script is exiting."
+            raise AdfError(errmsg)
+
+        #Search for first variable in var_list to get a time series file to read
+        #NOTE: it is assumed all the variables have the same dates!
+        ts_files = sorted(list(input_location.glob(f"{case_name}*.{var_list[0]}.*nc")))
```

Comment:
Don't need to convert to a list (as `sorted()` does that automatically):

```suggestion
        ts_files = sorted(input_location.glob(f"{case_name}*.{var_list[0]}.*nc"))
```